ARCHITECTURE.md

```diff
@@ -136,8 +261,224 @@ Mutation webhooks meant for intercepting user Kubernetes Resources, so they can
+
+### `pkg/webhook/mutation/pod`
+
+Main logic of the `MutationWebhook` mutating `Pods`. Does the mapping between the `Namespaces` and `DynaKube` from the `Namespace's` side.
+
+### `pkg/webhook/mutation/pod/metadata`
+
+Package that deals with necessary metadata for `Pods`. Sets up and prepares `Containers`, `Envs` and `Volumes` that are needed for mutating `Pods`.
+
+### `pkg/webhook/mutation/pod/oneagent`
+
+Package that deals with necessary metadata for `OneAgent` related metadata. Sets up for example `Volumes` which hold necessary files that are needed for `OneAgent` installation.
+
+### `pkg/webhook/util`
+
+Utils package that does operations like checking `EnvVars` in `Containers`.
+
+### `pkg/webhook/validation`
+
+Main logic for the `ValidationWebhook`. Deals with both `DynaKube` and `EdgeConnect`. Checks if the configuration of said Resources is valid and can be applied. Gives warnings or even errors if there is some misconfiguration.
+
+### `test`
+
+E2E testing code. Unit tests are NOT found here, they are in the same module that they are testing, as that is the Golang convention.
+
+### `test/features`
+
+Respective feature for each scenario. Those features are then called within the tests and ran through.
+
+### `test/helpers`
+
+Package that has helpers which are needed for the E2E tests.
+
+### `test/mocks`
+
+Package that has mocks which are needed for the E2E tests.
+
+### `test/project`
+
+Package that sets up the root directory that will be used within the E2E tests.
+
+### `test/scenarios`
+
+Package that has all scenarios that are tested by the E2E tests. Those scenarios have a `Test*` function that calls the features from the `feature` package and runs the tests.
+
+### `test/testdata`
+
+Has every data that is needed to run the tests. This reaches from `Deployments` up to `Secrets` to be able to install the `Operator`.
+
+---
+Find here some very basic visualizations of the Operator with its components:
+
+## Overall functionality
+
+```mermaid
+graph 
+    DK[fa:fa-dk Dynakube] --> WH(fa:fa-file Dynatrace-Webhook):::wh
+    DK[fa:fa-dk Dynakube] --> OP(fa:fa-op Dynatrace-Operator):::op
+    DK[fa:fa-dk Dynakube] --> CSI(fa:fa-csi CSI Driver):::csi
+    AG --> Dynatrace-Cluster
+    subgraph K8S Node
+    WH
+    OP -->|deploys| OA[OneAgent]:::oa
+    OP -->|deploys| AG[ActiveGate]:::ag
+    CSI --> P[Pods]:::pod
+    P --> AG
+    end
+
+    classDef op stroke:#f00
+    classDef wh stroke:#0f0
+    classDef csi stroke:#00f
+    classDef oa stroke:#ff0
+    classDef ag stroke:#000
+    classDef pod stroke:#f0f
+```
+
+## Operator
+
+```mermaid
+graph LR
+    subgraph Dynatrace Namespace
+    Dynatrace-Operator-Deployment -->|creates| RS[Dynatrace-Operator-ReplicaSet]:::rs
+    RS -->|manages| Pod[Dynatrace-Operator-Pod]:::pod
+    end
+
+    classDef rs stroke:#f00
+    classDef pod stroke:#0f0
+```
+
+## Webhook
+
+```mermaid
+graph LR
```

Comment:
I want to challenge the value these diagrams (webhook, CSI, OA, AG) add. They basically describe standard K8S functionality.

Reply:
I agree with you, I just did it that shallow because previously I simply did what the ticket told me (which was a little bit old) and just described the packages. Now, we decided to go for another approach and this is so to speak a "test" with mermaid to see if you guys like it. I will go deeper there.